pixeldata/src/encapsulation.rs

```diff
@@ -0,0 +1,207 @@
+use dicom_core::value::{Value, C};
+use dicom_core::DataDictionary;
+use dicom_object::mem::InMemFragment;
+use dicom_object::InMemDicomObject;
+
+#[derive(Debug)]
+pub struct EncapsulatedPixels {
+    bot: C<u32>,
+    current_offset: u32,
+    fragments: C<Vec<u8>>,
+    number_of_fragments: u32,
+}
+
+impl EncapsulatedPixels {
+    /// Add a single frame to EncapsulatedPixels
+    pub fn add_frame(&mut self, data: Vec<u8>) {
+        for fragment in fragment_frame(data, self.number_of_fragments) {
+            let size = fragment.len() as u32;
+            self.bot.push(self.current_offset);
+            self.fragments.push((*fragment).to_vec());
+            self.current_offset += size;
+        }
+    }
+
+    /// Creates an empty EncapsulatedPixels struct
+    pub fn new(number_of_fragments: u32) -> Self {
+        let number_of_fragments = if number_of_fragments > 0 {
+            number_of_fragments
+        } else {
+            1u32
+        };
+
+        EncapsulatedPixels {
+            bot: C::new(),
+            current_offset: 0,
+            fragments: C::new(),
+            number_of_fragments,
+        }
+    }
+}
+
+#[allow(clippy::from_over_into)]
+impl<D> Into<Value<InMemDicomObject<D>, InMemFragment>> for EncapsulatedPixels
+where
+    D: DataDictionary,
+    D: Clone,
+{
+    fn into(self) -> Value<InMemDicomObject<D>, InMemFragment> {
+        Value::PixelSequence {
+            offset_table: self.bot,
+            fragments: self.fragments,
+        }
+    }
+}

```

Comment:
This is not a good idea. The general recommendation is to implement `From` instead.

```suggestion
impl<D> From<EncapsulatedPixels> for Value<InMemDicomObject<D>, InMemFragment>
where
    D: DataDictionary,
    D: Clone,
{
    fn from(value: EncapsulatedPixels) -> Self {
        Value::PixelSequence {
            offset_table: value.bot,
            fragments: value.fragments,
        }
    }
}
```

Reply:
Great suggestion, I have implemented the suggested change.